src/sentry/models/grouphash.py

```diff
@@ -23,3 +24,38 @@ class Meta:
         app_label = 'sentry'
         db_table = 'sentry_grouphash'
         unique_together = (('project', 'hash'),)
+
+    @staticmethod
+    def fetch_last_processed_event_id(project_id, group_hash_ids):
+        prefix = 'last-processed-event:{}'.format(project_id)
+        with redis.clusters.get('default').map() as client:
+            results = map(
+                lambda group_hash_id: client.hget(
+                    '{}:{}'.format(prefix, group_hash_id % 16),
```

Comment:
I didn't test this but [conventional knowledge](https://redis.io/topics/data-types-intro#redis-hashes) and my assumption is that it's more efficient to store lots of small items in a hash than as independent keys, so that's what I did, but also tried to ensure that a project wouldn't put all of it keys in one bucket (heh heh) by modulo hashing it here. Open to changing this if somebody has a different idea.